interview-questions.md

```diff
@@ -9,59 +9,58 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: The keyword "this" refers to the object that is currently or specifically being invoked in a function.
 
-  Researched answer:
+  Researched answer: When calling a function as a method of the object, using "this" refers to that specific object being used or being referred to. 
 
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a Javascript library for building user interfaces. React contains code snippet that is reusable. It is component based. 
 
-  Researched answer:
+  Researched answer: React is a front-end Javascript Library that is used for building user interfaces. It can be used to create web applications and mobile apps. Because React is a library, it has reusuable code snippers and other UI components that allows it to be usuable. 
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
-
-  Researched answer:
+  Your answer: render() is required in a React class component. By using render(), it presents the specified HTML on the webpage. 
 
+  Researched answer: In React, render() method is the only required method in a class component. It describes how the webpage should display the HTML element and tags. We are able to read React props and states.
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is a hybrid between HTML and Javascript.  JSX fragments ( <> </>) can be used to turn HTML tags into react elements. The difference between HTML and JSX is that HTML is a markup language that allows for creating a structure of a page. But with JSX, it is like an extension where HTML can be used within Javascript code. 
 
-  Researched answer:
+  Researched answer: JSX is Javascript XML. JSX allows us to work with HTML in React within a Javascript code. It is easier for HTML to be changed or used. HTML is the standard langauge that makes the structure of a page. JSX on the other hand, allows developers to write HTML with their javascript code in React. 
```

Comment:
Another notable difference is the fact that JSX returns only a single element which is why we need React fragments when working with more than one element on a component.